### DISCUSSION TITLE
how to install for multiple instances on a single server.

### DISCUSSION THREAD

**Discussion by meotimdihia** [caching-tools/next-shared-cache#224] [Q&A]
Hi, I have multiple Next instances on a single server.
I feel it is overkill to use Redis. I wonder if we install `@neshca/cache-handler` to solve this problem.
While reading the documentation, I didn't get how multiple instances can share the same caching memory/or filesystem storage without Redis.

And can I continue to revalidate caching by this API `res.revalidate()`?
https://nextjs.org/docs/pages/building-your-application/data-fetching/incremental-static-regeneration#on-demand-revalidation

**Comment by better-salmon** [comment]
Hey, @meotimdihia! I apologize for the delay. By design, the `@neshca/cache-handler` is an API to implement different caching strategies. It works with the filesystem almost the same way as a native cache; you can create your own Handler to share the filesystem locally on your server.

Sharing the memory is a different task. You can try using [`@neshca/server`](https://caching-tools.github.io/next-shared-cache/server).

I tried to be as simple and concise as possible. If you have any additional questions, feel free to continue this discussion.
